api/server/server.go

```diff
@@ -662,7 +662,7 @@ func postImagesPush(eng *engine.Engine, version version.Version, w http.Response
 	} else {
 		// the old format is supported for compatibility if there was no authConfig header
 		if err := json.NewDecoder(r.Body).Decode(authConfig); err != nil {
-			return err
+			return fmt.Errorf("Bad parameters and missing X-Registry-Auth")
```

Comment:
I like the idea of a nice error message, but can we include the original error message text too since it might include a clue as to what was wrong with the data that was being decoded?